### PR TITLE
chore(j-s): Refactor upload

### PR DESCRIPTION
**Pull request by oddsson**
# Refactor upload

[Asana](https://app.asana.com/0/1199153462262248/1203374349779618/f)

## What

Refactor upload - Start using a new upload hook instead of the old `handleS3Upload` hook

## Why

The old hook used refs and had a lot of internal state. Was error prone and hard to debug.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Court/Indictments/CourtRecord/CourtRecord.tsx [hunk 1]
```diff
@@ -47,14 +57,42 @@ const CourtRecord: React.FC = () => {
...
+    }
+  }, [workingCase.caseFiles])
+
+  useMemo(() => {
+    setAllFilesUploaded(
+      displayFiles.every(
+        (file) => file.status === 'done' || file.status === 'error',
+      ),
+    )
+  }, [displayFiles, setAllFilesUploaded])
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/routes/Court/Indictments/CourtRecord/CourtRecord.tsx:80]: Consider refactoring this to not use state.
```suggestion
  const allFilesUploaded = useMemo(() => 
      displayFiles.every(
        (file) => file.status === 'done' || file.status === 'error',
      ), [displayFiles])
```

File: apps/judicial-system/web/src/routes/Prosecutor/Indictments/CaseFiles/CaseFiles.tsx [hunk 2]
```diff
@@ -39,17 +47,73 @@ const CaseFiles: React.FC = () => {
...
+  const { upload, remove } = useS3Upload(workingCase.id)
+
+  useEffect(() => {
+    if (workingCase.caseFiles) {
+      setDisplayFiles(workingCase.caseFiles.map(mapCaseFileToUploadFile))
+    }
+  }, [workingCase.caseFiles])
+
+  useMemo(() => {
+    setAllFilesUploaded(
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/routes/Prosecutor/Indictments/CaseFiles/CaseFiles.tsx:62]: Same here, consider removing the need for state.

**Review by Arnarkari93 — CHANGES_REQUESTED** [review]
LGMT, just one suggestion

File: apps/judicial-system/web/src/routes/Court/Indictments/CourtRecord/CourtRecord.tsx (hunk 1 above)
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Court/Indictments/CourtRecord/CourtRecord.tsx]: Awesome! Thanks 🙏

**Review by Arnarkari93 — APPROVED** [review]

**Pull request merged**